### PR TITLE
Resolve Date Time Issues

### LINKED ISSUE #134: Resolve Date Time Issues
# Resolve Date Time Issues
Fix the date time field for event creation to ensure that it fills in properly for existing events.

## Why?
Currently, the date time field does not auto-populate in the event creation form as it should.

## Acceptance Criteria
- [ ] Fix the date time field for event creation
- [ ] Ensure that the "Start Time" label appears on the date time field

### PR DESCRIPTION
**Pull request by ajaygandecha**
This pull request fixes major issues with how dates worked on the Event frontend of the CSXL application.

## Major Changes
- Fixed error where the "edit date" field for an event would not show the correct time upon editing.
- Fixed error where dates sometimes would display correctly.
- Re-fix error where past events would show on the events page.
- These errors were fixed by using the DatePipe to convert dates to the exact format that `matInput` expects for dates.

## Testing
- These features were tested on the frontend using `honcho start`.

*This PR resolves #134*

### CODE REVIEW

File: frontend/src/app/event/event-filter/event-filter.pipe.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ export class EventFilterPipe implements PipeTransform {
       return a.time.getTime() - b.time.getTime();
     });
 
+    // Filter out past dates
+    events = events.filter((e: Event) => {
+      return e.time.getTime() > new Date().getTime()
```
Comment (KrisJordan) [inline frontend/src/app/event/event-filter/event-filter.pipe.ts:29]: We should still be doing pagination at the API level for viewing past events and only typically loading future events.

**Review by KrisJordan — APPROVED** [review]
LGTM with the future goal of event filtering and pagination being server-side.

**Comment by KrisJordan** [comment]
Thanks for getting to the bottom of this mystery!

**Comment by ajaygandecha** [comment]
Sounds great - that ticket will be taken up next, just wanted to put a 🩹 on it while that was done!

**Pull request merged**